core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp

```diff
@@ -658,8 +658,11 @@ struct CudaParallelLaunchImpl<
       // Invoke the driver function on the device
       base_t::invoke_kernel(driver, grid, block, shmem, cuda_instance);
 
-#if defined(KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK)
+      // check any launch error (synchronous check)
```

Comment:
@crtrott brought up that this was a conscious decision in the past because the `cudaGetLastError()` call costed a few microseconds.
I suppose the launch latency benchmarks should confirm that this is a bad idea.  (Please check and post the results in the description)
We could add comments in the code to capture this.

Reply:
We discussed that we want to keep everything as-is for now here (making sure we get consistent behavior across all backends and document it).